core/src/main/java/jenkins/I18n.java

```diff
@@ -95,7 +95,7 @@ public HttpResponse doResourceBundle(StaplerRequest request) {
         String variant = request.getParameter("variant");
         
         try {
-            Locale locale = Locale.getDefault();
+            Locale locale = request.getLocale();
```

Comment:
So long as `request.getLocale()` guarantees to return something. Might be worth checking and using `Locale.getDefault()` if `null` is returned.